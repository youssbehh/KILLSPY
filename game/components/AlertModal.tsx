import React from 'react';
import { Modal, View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { Button } from '@rneui/themed';

interface AlertModalProps {
    visible: boolean;
    text1: string; // Message à afficher dans le modal
    text2?: string;
    button1: string; // Titre du premier bouton
    onPress1: () => void; // Fonction à appeler pour le premier bouton
    button1Style?: object;
    disabled1?: boolean;
    button2?: string; // Titre du deuxième bouton (optionnel)
    onPress2?: () => void; // Fonction à appeler pour le deuxième bouton (optionnel)
}

const AlertModal: React.FC<AlertModalProps> = ({
    visible,
    text1,
    text2,
    button1,
    onPress1,
    button1Style,
    disabled1,
    button2,
    onPress2,
}) => {
    return (
        <Modal visible={visible} transparent>
            <View style={styles.overlay}>
                <View style={styles.modalContainer}>
                    <Text style={styles.title}>{text1}</Text>
                    {text2 && (
                        <Text style={styles.titleopt}>{text2}</Text>
                    )}
                        <Button style={[styles.button, button1Style]} onPress={disabled1 ? () => {} : onPress1} disabled={disabled1}>{button1}</Button>
                    {button2 && onPress2 && (
                        <Button style={styles.button} onPress={onPress2}>{button2}</Button>
                    )}
                </View>
            </View>
        </Modal>
    );
};

const styles = StyleSheet.create({
    overlay: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(0,0,0,0.5)',
    },
    modalContainer: {
        width: 300,
        padding: 20,
        backgroundColor: 'white',
        borderRadius: 10,
    },
    title: {
        padding: 5,
        fontSize: 18,
        fontWeight: 'bold',
    },
    titleopt: {
        padding: 5,
        fontSize: 18,
    },
    button: {
        padding: 10,
    },
    closeButton: {
        padding: 10,
        backgroundColor: 'gray',
        borderRadius: 5,
        marginTop: 10,
    },
});

export default AlertModal;