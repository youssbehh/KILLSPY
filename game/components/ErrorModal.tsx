import React from 'react';
import { Modal, View, Text, StyleSheet } from 'react-native';
import { Button } from '@rneui/themed';

interface ErrorModalProps {
  visible: boolean;
  title: string;
  message: string;
  onClose: () => void;
}

const ErrorModal: React.FC<ErrorModalProps> = ({ visible, title, message, onClose }) => {
  return (
    <Modal
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <View style={styles.overlay}>
        <View style={styles.modalContainer}>
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.message}>{message}</Text>
          <Button color='error' onPress={onClose}>Retour</Button>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
  },
  modalContainer: {
    width: '80%',
    padding: 20,
    backgroundColor: 'white',
    borderRadius: 10,
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  message: {
    fontSize: 16,
    marginBottom: 20,
  },
});

export default ErrorModal;