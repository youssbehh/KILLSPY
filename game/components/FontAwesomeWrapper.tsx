import React from 'react';
import { Platform } from 'react-native';
import { FontAwesomeIcon as WebIcon } from '@fortawesome/react-fontawesome';
import { FontAwesomeIcon as NativeIcon } from '@fortawesome/react-native-fontawesome';
import { IconDefinition, SizeProp } from '@fortawesome/fontawesome-svg-core';

interface FontAwesomeWrapperProps {
  icon: IconDefinition;
  color?: string;
  size?: number;
}

export const FontAwesomeWrapper: React.FC<FontAwesomeWrapperProps> = ({ icon, color, size }) => {
  if (Platform.OS === 'web') {
    return <WebIcon icon={icon} color={color} size={(size?.toString() || 'lg') as SizeProp} />;
  }
  return <NativeIcon icon={icon} color={color} size={size} />;
};