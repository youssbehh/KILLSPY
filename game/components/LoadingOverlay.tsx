import React from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

interface LoadingOverlayProps {
  isLoading: boolean; // Définir le type de isLoading
}

const LoadingOverlay: React.FC<LoadingOverlayProps> = ({ isLoading }) => {
  if (!isLoading) return null; // Ne rien afficher si isLoading est faux

  return (
    <View style={styles.loadingOverlay}>
      <ActivityIndicator size="large" color="#007AFF" />
    </View>
  );
};

const styles = StyleSheet.create({
  loadingOverlay: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'rgba(0, 0, 0, 0.5)', // Assombrit l'arrière-plan
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1000, // Assurez-vous que l'overlay est au-dessus des autres éléments
  },
});

export default LoadingOverlay;