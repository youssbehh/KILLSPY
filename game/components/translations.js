   const translations = {
    0: { // Français
      1: 'Bienvenue',
      2: 'Accueil',
      3: 'Jouer',
      4: 'Boutique',
      5: 'Mode de jeux',
      6: 'Profil',
      7: 'Options',
      8: 'Inventaire',
      9: 'Amis',
      10: 'Liste des agents',
      11: 'Bienvenue, agent',
      12: 'VS Ordinateur',
      13: 'Partie Rapide',
      14: 'Compétitif',
      15: 'Événements',
      16: 'Classement',
      17: 'Rang',
      18: 'Liste d\'amis',
      19: 'Audio',
      20: 'Mon compte',
      21: 'Sécurité',
      22: 'Divers',
      23: 'Musique',
      24: 'Effets audio',
      25: 'Langues',
      26: 'Français',
      27: 'Anglais',
      28: 'Support',
      29: 'À propos',
      30: 'Mentions légales',
      31: 'Politique de confidentialité',
      32: 'Match nul',
      33: 'Recherche de partie en cours',
      34: 'Partie trouvée',
      35: 'Annuler',
      36: 'Vous avez gagné',
      37: 'Vous avez perdu',
      38: 'Recommencer',
      39: 'Quitter',
      40: 'Bouclier',
      41: 'Tirer',
      42: 'Recharger',
      43: 'Continuer',
      44: 'Quitter la partie ?',
      45: 'Êtes-vous sûr de vouloir quitter ?',
      46: 'Changer de nom d\'utilisateur',
      47: 'Nous contacter',
      48: 'Information',
      49: 'Dernière parties',
      50: 'Jouer en tant qu\'invité',
      51: 'Déconnexion',
      52: 'Email ou nom d\'utilisateur',
      53: 'Mot de passe',
      54: 'Mot de passe oublié ?',
      55: 'Créer un compte',
      56: 'Connexion',
      57: 'Se souvenir de moi'
    },
    1: { // Anglais
      1: 'Welcome',
      2: 'Home',
      3: 'Play',
      4: 'Shop',
      5: 'Gamemode',
      6: 'Profile',
      7: 'Options',
      8: 'Inventory',
      9: 'Friends',
      10: 'Agents list',
      11: 'Welcome, agent',
      12: 'VS Bot',
      13: 'Normal game',
      14: 'Ranked',
      15: 'Events',
      16: 'Leaderboard',
      17: 'Rank',
      18: 'Friends list',
      19: 'Audio',
      20: 'My account',
      21: 'Security',
      22: 'Others',
      23: 'Music',
      24: 'SFX',
      25: 'Languages',
      26: 'French',
      27: 'English',
      28: 'Support',
      29: 'About',
      30: 'Legal mentions',
      31: 'Privacy policy',
      32: 'Draw',
      33: 'Searching for a game',
      34: 'Game found',
      35: 'Cancel',
      36: 'You won',
      37: 'You lost',
      38: 'Restart',
      39: 'Quit',
      40: 'Shield',
      41: 'Shoot',
      42: 'Reload',
      43: 'Continue',
      44: 'Quit game ?',
      45: 'Are you sure you want to quit ?',
      46: 'Change username',
      47: 'Contact us',
      48: 'Information',
      49: 'Latest Games',
      50: 'Play with guest account',
      51: 'Logout',
      52: 'Email or username',
      53: 'Password',
      54: 'Forgot password ?',
      55: 'Create an account',
      56: 'Login',
      57: 'Remember me'
    },
  };

  export default translations;