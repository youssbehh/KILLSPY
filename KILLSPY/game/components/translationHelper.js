      import translations from './translations';

      export const motTraduit = (langIndex, wordIndex) => {
        return translations[langIndex][wordIndex] || '';
      };