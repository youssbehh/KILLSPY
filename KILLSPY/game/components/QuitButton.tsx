import React from 'react';
import { StyleSheet, Pressable, Alert } from 'react-native';
import { Text } from '@/components/Themed';
import { router } from 'expo-router';
import { motTraduit } from '@/components/translationHelper';
import { useLanguageStore } from '@/store/languageStore';

interface QuitButtonProps {
    redirectPath?: any;
}

export default function QuitButton({ redirectPath = '/(tabs)' }: QuitButtonProps) {
    const { langIndex } = useLanguageStore();

    const handleQuit = () => {
        Alert.alert(
            motTraduit(langIndex, 44), // "Quitter la partie ?"
            motTraduit(langIndex, 45), // "Êtes-vous sûr de vouloir quitter ?"
            [
                {
                    text: motTraduit(langIndex, 35), // "Annuler"
                    style: 'cancel',
                },
                {
                    text: motTraduit(langIndex, 39), // "Quitter"
                    style: 'destructive',
                    onPress: () => router.replace(redirectPath),
                },
            ],
            { cancelable: true }
        );
    };

    return (
        <Pressable 
            onPress={handleQuit}
            style={({ pressed }) => [
                styles.quitButton,
                pressed && styles.quitButtonPressed
            ]}
        >
            <Text style={styles.quitButtonText}>✕</Text>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    quitButton: {
        padding: 10,
        marginRight: 10,
    },
    quitButtonPressed: {
        opacity: 0.7,
    },
    quitButtonText: {
        fontSize: 20,
        color: '#FF3B30',
        fontWeight: 'bold',
    },
});
