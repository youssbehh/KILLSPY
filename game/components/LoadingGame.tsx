import React, { useEffect, useState } from 'react';
import { StyleSheet, ActivityIndicator, Pressable } from 'react-native';
import { Text, View } from '@/components/Themed';
import { motTraduit } from './translationHelper';
import { useLanguageStore } from '@/store/languageStore';
import { router, useRouter } from 'expo-router';
import socket from '@/services/socket';

interface LoadingGameProps {
  gameMode: string;
  onGameFound: () => void;
}

interface GameFoundData {
    opponentId: string; // Define the type for opponentId
}

const dotAnimation = ['.', '..', '...'];

export default function LoadingGame({ gameMode, onGameFound }: LoadingGameProps) {
    const { langIndex } = useLanguageStore();
    const router = useRouter();
    const [dotIndex, setDotIndex] = useState(0);
    const [isGameFound, setIsGameFound] = useState(false);

    useEffect(() => {
        const interval = setInterval(() => {
            setDotIndex((prevIndex) => (prevIndex + 1) % dotAnimation.length);
        }, 500);
        return () => clearInterval(interval);
    }, []);

    useEffect(() => {
        if (gameMode === motTraduit(langIndex, 12) && !isGameFound) { // VS Ordinateur
            const timeout = setTimeout(() => {
                setIsGameFound(true);
                onGameFound();
                router.push('/play/bot');
            }, 3000);
            return () => clearTimeout(timeout);
        }
    }, [gameMode, langIndex, onGameFound]);

    useEffect(() => {
        socket.on('gameFound', (data: GameFoundData) => {
            const { opponentId } = data;
            console.log('Jeu trouvÃ© avec l\'opposant:', opponentId);
            setIsGameFound(true);
            onGameFound();
            router.push('/play/partieRapide');
        });

        return () => {
            socket.off('gameFound');
        };
    }, []);

    const handleCancelSearch = () => {
        socket.emit('cancelSearch');
        router.back();
    };

    return (
        <View style={styles.container}>
            <Text style={styles.title}>{gameMode}</Text>
            <ActivityIndicator size="large" color="#0000ff" style={styles.loader} />
            <Text style={styles.loadingText}>
                {isGameFound ? motTraduit(langIndex, 34) : motTraduit(langIndex, 33) + dotAnimation[dotIndex]}
            </Text>
            {!isGameFound && (
                <Pressable onPress={handleCancelSearch}>
                    <Text style={styles.backButton}>{motTraduit(langIndex, 35)}</Text>
                </Pressable>
            )}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        fontSize: 24,
        fontWeight: 'bold',
        marginBottom: 20,
    },
    loader: {
        marginVertical: 20,
    },
    loadingText: {
        fontSize: 16,
    },
    backButton: {
        fontSize: 16,
        color: '#007AFF',
        marginTop: 20,
    },
});