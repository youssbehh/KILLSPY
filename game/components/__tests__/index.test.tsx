import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { useRouter } from 'expo-router';
import LoginFormScreen from '../../app/index';
import { motTraduit } from '@/components/translationHelper';

// Mock de useRouter
const mockReplace = jest.fn();
jest.mock('expo-router', () => ({
  useRouter: () => ({
    replace: mockReplace,
  }),
}));

// Mock de la traduction
jest.mock('@/components/translationHelper', () => ({
  motTraduit: jest.fn((langIndex, id) => `traduit-${id}`),
}));

// Mock du store de langue
jest.mock('../../store/languageStore', () => ({
  useLanguageStore: () => ({
    langIndex: 0,
  }),
}));

describe('LoginFormScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('affiche le titre KILLSPY', () => {
    const { getByText } = render(<LoginFormScreen />);
    expect(getByText('KILLSPY')).toBeTruthy();
  });

  it('affiche le texte traduit avec motTraduit', () => {
    const { getByText } = render(<LoginFormScreen />);
    expect(getByText('traduit-50')).toBeTruthy();
    expect(motTraduit).toHaveBeenCalledWith(0, 50);
  });

  it('navigue vers la page gamechoice lorsqu’on appuie sur le texte', () => {
    const { getByText } = render(<LoginFormScreen />);
    fireEvent.press(getByText('traduit-50'));

    expect(mockReplace).toHaveBeenCalledWith('/(tabs)/gamechoice');
  });

  it('crée un snapshot du composant LoginFormScreen', () => {
    const tree = render(<LoginFormScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
