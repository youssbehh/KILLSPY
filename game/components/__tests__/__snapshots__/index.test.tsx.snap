// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormScreen cr√©e un snapshot du composant LoginFormScreen 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontSize": 32,
          "fontWeight": "bold",
        },
      ]
    }
  >
    KILLSPY
  </Text>
  <Text
    onPress={[Function]}
    style={
      [
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    traduit-50
  </Text>
</View>
`;
